cpp/tests/io/orc_test.cpp

```diff
@@ -1454,35 +1454,39 @@ TEST_F(OrcWriterTest, DecimalOptionsNested)
   auto const num_rows = 100;
 
   auto dec_vals  = random_values<int32_t>(num_rows);
-  auto keys_data = cudf::detail::make_counting_transform_iterator(0, [&](auto i) {
+  auto dec1_data = cudf::detail::make_counting_transform_iterator(0, [&](auto i) {
     return numeric::decimal64{dec_vals[i], numeric::scale_type{2}};
   });
-  auto vals_data = cudf::detail::make_counting_transform_iterator(0, [&](auto i) {
+  auto dec2_data = cudf::detail::make_counting_transform_iterator(0, [&](auto i) {
     return numeric::decimal128{dec_vals[i], numeric::scale_type{2}};
   });
   auto validity  = cudf::detail::make_counting_transform_iterator(0, [](auto i) { return true; });
-  column_wrapper<numeric::decimal64> keys_col{keys_data, keys_data + num_rows, validity};
-  column_wrapper<numeric::decimal128> vals_col{vals_data, vals_data + num_rows, validity};
+  column_wrapper<numeric::decimal64> dec1_col{dec1_data, dec1_data + num_rows, validity};
+  column_wrapper<numeric::decimal128> dec2_col{dec2_data, dec2_data + num_rows, validity};
+  auto child_struct_col = cudf::test::structs_column_wrapper{dec1_col, dec2_col};
 
-  auto struct_col = cudf::test::structs_column_wrapper({keys_col, vals_col}).release();
+  auto int_vals                   = random_values<int32_t>(num_rows);
+  column_wrapper<int32_t> int_col = {int_vals.begin(), int_vals.end(), validity};
```

Comment:
IMO most changes in this PR may leverage the `auto` keyword to shorten the code.

Reply:
Could you give me an example? I can't find a way to use auto here to shorten the code.

Reply:
Sorry I was wrong. I meant using type aliasing instead. For example, instead of having:
```
  column_wrapper<numeric::decimal64> dec1_col{dec1_data, dec1_data + num_rows, validity};
  column_wrapper<numeric::decimal128> dec2_col{dec2_data, dec2_data + num_rows, validity};
  column_wrapper<int32_t> int_col = {int_vals.begin(), int_vals.end(), validity};
```
we can have something like
```
  using dec64_col = column_wrapper<numeric::decimal64>;
  ...

  dec64_col dec1_col{dec1_data, dec1_data + num_rows, validity};
  dec128_col dec2_col{dec2_data, dec2_data + num_rows, validity};
  int32_col int_col = {int_vals.begin(), int_vals.end(), validity};
```

The aliased types may be used multiple times. Even they are used just one, the shorter names will help the code look cleaner.

Reply:
Added aliases. Most of the PR now is unrelated to the added test. Let me know if I should split the clean up into a separate PR.